go/runtime/host/helpers.go

```diff
@@ -119,6 +127,34 @@ func (r *richRuntime) Query(
 	return resp.RuntimeQueryResponse.Data, nil
 }
 
+// Implements RichRuntime.
+func (r *richRuntime) QueryBatchLimits(
+	ctx context.Context,
+	rb *block.Block,
+	lb *consensus.LightBlock,
+	epoch beacon.EpochTime,
+) (map[transaction.Weight]uint64, error) {
+	if rb == nil {
+		return nil, ErrInvalidArgument
+	}
+
+	resp, err := r.Query(ctx, rb, lb, epoch, protocol.MethodQueryBatchWeightLimits, nil)
+	switch {
+	case errors.Is(err, protocol.ErrQueryNotSupported):
+		// Runtime does not implement custom batch limits.
+		return make(map[transaction.Weight]uint64), nil
+	case err != nil:
+		return nil, err
+	default:
+	}
+
+	var weightLimits map[transaction.Weight]uint64
+	if err = cbor.Unmarshal(resp, &weightLimits); err != nil {
+		return nil, errors.WithContext(fmt.Errorf("%v: %w", ErrInternal, err), "malformed runtime response")
```

Comment:
Wow that's a weird construction, should you do something like this instead:
```suggestion
		return nil, errors.WithContext(ErrInternal, fmt.Sprintf("malformed runtime response: %v", err))
```
Maybe we could even make `WithContext` take variadic arguments so it would work without `fmt.Sprintf`.